### DISCUSSION TITLE
[Q] Multi tenancy (multiple sites/projects)

### DISCUSSION THREAD

**Discussion by binaryfire** [roadrunner-server/roadrunner#757] [Q&A] [answered]
Hi guys,

Is there any way to use a single instance of Roadrunner on a server with multiple Laravel projects (sites)? Each one will have a different domain / SSL cert. Perhaps in conjunction with an NGINX reverse proxy or something?

If not, I'd love to see a feature like this.

**Comment by rustatian** [comment, marked as answer]
Hey @binaryfire. Unfortunately, this is not possible at the time. I don't think, that this is even possible to implement via 1 RR instance. Because it can be easily solved by running multiply RR instances (bash script). Only 1 thing you'll need to maintain - RR configuration files.
```bash
#/usr/bin/bash
while true; ./rr serve -c $1; sleep 1; end
```
Just replace `true` with the number of needed instances and provide paths to the configuration files.

**Reply by lucatacconi** [reply]
so, just to understand if I have correctly understood how to use RoadRunner: starting from an Nginx + PHP server that serves different virtuahost the correct way is to have different RR process, one for single virtualhost. Is this scenario correct? Could there be problems running different instances of rr configured by different yml config files on the same machine?

Thank you in advance guys.

**Reply by rustatian** [reply]
Hey @lucatacconi 👋🏻 . There are no problems with running different RR instances with different configurations. Just keep in mind that the resources of your server is limited :)

**Reply by rustatian** [reply]
> Is this scenario correct?

You may have 1 RR instance and redirect all traffic to it. Then you may route the traffic inside your application.
For example, if you send a request to the `http://your.site/foo/bar` RR will handle it (because it's mounted to the root (`/`)), and you may route that request inside your app (you'll get the `/foo/bar` path inside the worker).
